### PR TITLE
Drop Python 2

### PR DESCRIPTION
**Pull request by pgjones**
I'm not sure what release this will be in, 1, 1.1, 2?

### CODE REVIEW

**Comment by davidism** [comment]
Well, I've only been looking forward to doing this for 2 years, but ok, you get the pr 😉

This will be in 2, so I can't merge it yet.

File: tests/test_datastructures.py [hunk 1]
```diff
@@ -795,23 +755,15 @@ def test_to_wsgi_list(self):
         h = self.storage_class()
         h.set(u"Key", u"Value")
         for key, value in h.to_wsgi_list():
-            if PY2:
-                strict_eq(key, b"Key")
-                strict_eq(value, b"Value")
-            else:
-                strict_eq(key, u"Key")
-                strict_eq(value, u"Value")
+            strict_eq(key, u"Key")
```
Comment (davidism) [inline tests/test_datastructures.py]: Can get rid of at least `u""`, and probably `strict_eq`.

**Comment by davidism** [comment]
Can use f-strings everywhere. Can drop u prefix.

I'd like to investigate moving away from supporting both bytes *and* text as arguments everywhere, as Python 3 should have much clearer delineation between where those types should be present. In the past there was an issue about Pallets being much slower after the 2/3 compat transition, that was caused by the constant type checking and conversion that had to be added. This can be a separate ticket though.

File: src/werkzeug/_internal.py [hunk 2]
```diff
@@ -64,6 +64,69 @@ def __reduce__(self):
...
+    raise TypeError("Expected bytes")
+
+
+def _to_native(x, charset=sys.getdefaultencoding(), errors="strict"):  # noqa
+    if x is None or isinstance(x, str):
+        return x
+    return x.decode(charset, errors)
+
+
+def _to_unicode(
```
[the remark this answers is not in the document]
Comment (pgjones) [reply src/werkzeug/_internal.py]: Done, I went for `_to_str`

File: src/werkzeug/_compat.py [hunk 3]
```diff
@@ -1,224 +1,3 @@
...
-        return None
-    if not isinstance(x, bytes):
-        return text_type(x)
-    if charset is None and allow_none_charset:
-        return x
-    return x.decode(charset, errors)
-
-
 try:
     from os import fspath
```
[the remark this answers is not in the document]
Comment (pgjones) [reply src/werkzeug/_compat.py]: Done

File: tests/test_datastructures.py (hunk 1 above)
Comment (pgjones) [reply tests/test_datastructures.py]: Done.

**Comment by and-semakin** [comment]
@davidism, note: you can't use f-strings, because of Python 3.5. f-string syntax is supported since 3.6.

**Comment by davidism** [comment]
@and-semakin we're supporting 3.6+: https://palletsprojects.com/blog/ending-python2-support/

**Comment by davidism** [comment]
I did some cleanup of the helpers we kept from `_compat`.

Replaced `to_native` with `to_str`, or removed the call completely where possible. I only looked at uses of `to_native`, there are plenty of places where `to_str` or `to_bytes` could be removed. There's also inconsistency in whether a charset is passed to them, I feel like it should default to `utf-8` instead of the system encoding, since otherwise things are just working by accident.

Removed `reraise`, as there's only one syntax we need to support now.

Removed the `strict_eq` test helper, as it was only for dealing with Python 2 casing unicode to str during comparisons.

**Comment by davidism** [comment]
I've tried to re-run the Windows job 3 times now, the reloader tests just refuse to cooperate today. Should probably skip most of them on Windows, they're unusable right now.

**Pull request merged**